src/rpc/server.cpp

```diff
@@ -370,14 +368,12 @@ static const CRPCCommand vRPCCommands[] =
     { "rawtransactions",    "fundrawtransaction",     &fundrawtransaction,     false },
 #endif
 
-#ifdef ENABLE_ADDRESS_INDEXING
     /* Address index */
     { "addressindex",       "getaddressmempool",      &getaddressmempool,      true  },
     { "addressindex",       "getaddressutxos",        &getaddressutxos,        false },
     { "addressindex",       "getaddressdeltas",       &getaddressdeltas,       false },
     { "addressindex",       "getaddresstxids",        &getaddresstxids,        false },
     { "addressindex",       "getaddressbalance",      &getaddressbalance,      false },
```

Comment:
This is a minor comment, feel free to ignore.
If we want to keep the behavior consistent with what we have now, we should filter these RPCs form the list produced in `CRPCTable::help`, as it happens for the `hidden` category.

But maybe the new behavior is preferable, if we have appropriate warnings in place in the help messages, as suggested above.

Reply:
As far as I can see all the rpc methods in the "addressindex" category return an error if related indexes are not enabled

Reply:
Yes I agree. I'm suggesting to update the help msg for the followings RPCs to state it clearly that some flags must be enabled for them to function correctly:
- getspentinfo (`spentindex`)
- getblockhashes (`timestampindex`)
- getblockdeltas (which misses the help msg completely, `spentindex`)

Reply:
Addressed in [537f382](https://github.com/HorizenOfficial/zen/pull/499/commits/537f382619e2a423626689b6f3d5c4b60cf6a2e6)

Please complete help description for getblockdeltas

Reply:
Pfa the description for getblockdeltas. Notice that this is mostly borrowed from zcash, and checked for consistency with our codebase.

[getblockdeltas.txt](https://github.com/HorizenOfficial/zen/files/10448602/getblockdeltas.txt)

Reply:
Thank you. Done in 1db7a1b

Reply:
Thank you